src/main/java/com/breskul/bibernate/util/EntityUtil.java

```diff
@@ -251,27 +330,121 @@ public static Class<?> getEntityCollectionElementType(Field field) {
   }
 
 
-  //todo check if we should use getClassColumnFields instead this one
-  public static <T> List<Field> getEntityColumns(Class<? extends T> entityClass) {
-    return Arrays.stream(entityClass.getDeclaredFields())
-        .filter(field -> field.isAnnotationPresent(Column.class))
-        .collect(Collectors.toList());
+  /**
+   * Retrieves an instance of the Collection class based on the given Field representing a
+   * collection.
+   *
+   * @param collectionField - The Field representing the collection
+   * @return An instance of the Collection class
+   * @throws IllegalArgumentException if the collection type is unsupported
+   */
+  public static Collection<Object> getCollectionInstance(Field collectionField) {
+    var collectionClass = collectionField.getType();
+
+    if (collectionClass.isAssignableFrom(List.class)) {
+      return new ArrayList<>();
+    }
+
+    if (collectionClass.isAssignableFrom(Set.class)) {
+      return new HashSet<>();
+    }
+
+    throw new IllegalArgumentException(
+        "Unsupported collection: " + collectionClass); // change exception and more clear msg?
+  }
+
+
+  /**
+   * Reads the value of a field on the given entity object.
+   *
+   * @param entity  - The entity object
+   * @param idField - The field to read the value from
+   * @return The value of the field
+   * @throws EntityParseException if failed to access the field due to illegal access
+   */
+  public static Object readFieldValue(Object entity, Field idField) {
+    try {
+      idField.setAccessible(true);
+      return idField.get(entity);
+    } catch (IllegalAccessException e) {
+      throw new EntityParseException(
+          "Failed to access field '" + idField.getName() + "' of entity type '" + entity.getClass()
+              .getName() + "': Illegal access");
+    }
+  }
```

Comment:
This method is present in `ReflectionUtil` class, please merge `develop` branch into your branch first and use that method.

Reply:
It was added by mistake during resolution of merge conflicts, will remove

Reply:
removed